### PR TITLE
K8SPXC-435 Do not show any passwords in log

### PR DESCRIPTION
**Pull request by hors**
[K8SPXC-435](https://jira.percona.com/browse/K8SPXC-435)

### CODE REVIEW

**Comment by JNKPercona** [comment]
PXC operator docker - perconalab/percona-xtradb-cluster-operator:PR-579-dec6b148

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| storage | passed |
| big-data | passed |
| haproxy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| security-context | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |

File: build/pxc-entrypoint.sh [hunk 1]
```diff
@@ -233,7 +236,10 @@ if [ -z "$CLUSTER_JOIN" ] && [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 
 	if [ ! -d "$DATADIR/mysql" ]; then
 		file_env 'MYSQL_ROOT_PASSWORD'
+
+		{ set +x; } 2>/dev/null
 		if [ -z "$MYSQL_ROOT_PASSWORD" -a -z "$MYSQL_ALLOW_EMPTY_PASSWORD" -a -z "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then
+		set -x
```
Comment (delgod) [inline build/pxc-entrypoint.sh:242]: logging become enabled back only if condition is true

File: build/pxc-entrypoint.sh [hunk 2]
```diff
@@ -276,7 +282,9 @@ if [ -z "$CLUSTER_JOIN" ] && [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 			mysql_tzinfo_to_sql /usr/share/zoneinfo | sed 's/Local time zone must be set--see zic manual page/FCTY/' | "${mysql[@]}" mysql
 		fi
 
+		{ set +x; } 2>/dev/null
 		if [ ! -z "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then
+		set -x
```
Comment (delgod) [inline build/pxc-entrypoint.sh:287]: the same

File: build/pxc-entrypoint.sh [hunk 3]
```diff
@@ -341,7 +351,9 @@ if [ -z "$CLUSTER_JOIN" ] && [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 
 		file_env 'MYSQL_USER'
 		file_env 'MYSQL_PASSWORD'
+		{ set +x; } 2>/dev/null
 		if [ "$MYSQL_USER" -a "$MYSQL_PASSWORD" ]; then
+		set -x
```
Comment (delgod) [inline build/pxc-entrypoint.sh:356]: the same

File: build/pxc-entrypoint.sh [hunk 4]
```diff
@@ -357,7 +369,9 @@ if [ -z "$CLUSTER_JOIN" ] && [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 			process_init_file "$f" "${mysql[@]}"
 		done
 
+		{ set +x; } 2>/dev/null
 		if [ ! -z "$MYSQL_ONETIME_PASSWORD" ]; then
+		set -x
```
Comment (delgod) [inline build/pxc-entrypoint.sh:374]: the same

**Review by delgod — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Pull request merged**